code/hud/hudtargetbox.cpp

```diff
@@ -1566,43 +1566,51 @@ void get_turret_subsys_name(ship_weapon *swp, char *outstr)
 
 	//WMC - find the first weapon, if there is one
 	if (swp->num_primary_banks || swp->num_secondary_banks) {
-		auto flags = turret_weapon_aggregate_flags(swp);
-
-		// check if beam or flak using weapon flags
-		if (flags[Weapon::Info_Flags::Beam]) {
-			sprintf(outstr, "%s", XSTR("Beam turret", 1567));
-		} else if (flags[Weapon::Info_Flags::Flak]) {
-			sprintf(outstr, "%s", XSTR("Flak turret", 1566));
+		int weaponid;
+		if (swp->num_primary_banks) {
+			weaponid = swp->primary_bank_weapons[0];
+		}
+		else {
+			weaponid = swp->secondary_bank_weapons[0];
+		}
```

Comment:
Perhaps it should search through all the banks instead of just checking the first ones?